### ISSUE TITLE
Should we convert `logging` timestamps from UTC to e.g CET/CEST?

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1694]
Right now, log timestamps are in the same locale as the machine - often UTC.

This new feature would also come with a new optional configuration variable defining the required timezone.

Note: this is _not_ about database timestamps, which are stored in UTC and sent to the client as timezone-aware timestamps, so that the client can convert them as needed.

**Comment by tcompa** [comment]
Thanks @ychiucco for #1719 - this is a very useful example of how it could/should work. We are not merging it right now, as it would require some further work (mainly to look for something less custom) and there are more pressing priorities. But it's a valid starting point for when we come back to this issue.

**Comment by ychiucco** [comment]
https://github.com/fractal-analytics-platform/fractal-server/blob/5abf1e666fa8281f7c341d5867ba19dfaac31f62/fractal_server/gunicorn_fractal.py#L21-L25

If we do not define `datefmt`, `asctime` will automatically contains milliseconds.
With a custom `datefmt`, milliseconds should be added using `%(msecs)`.

**Comment by jluethi** [comment]
Great work! Yes, would be useful to get local time in the server logs to avoid confusion :)

**Comment by jluethi** [comment]
The server job page shows me the runtimes in local time, but when I open the workflow.log, it shows logs in UTC.

## Example UZH (Pelkmans lab):
<img width="1104" alt="Screenshot 2024-11-15 at 16 55 56" src="https://github.com/user-attachments/assets/324e488c-bc36-4da7-b49e-860025af4934">

And in the log:
```
2024-09-19 14:39:05,839 - WF41_job453 - INFO - Start execution of workflow "Multiplexing_test"; 
```

## Example FMI:
<img width="1063" alt="Screenshot 2024-11-15 at 16 51 35" src="https://github.com/user-attachments/assets/c6566d01-8f45-49e2-900e-7e53cf0ec458">

And in the log:
```
2024-11-14 13:44:21,264 - WF431_job1812 - INFO - Start execution of workflow "Demo workflow"; more logs at 
```

----

Admittedly, the tricky part here is that these logs are written server-side during workflow execution. And they can't be adapted live on the front-end depending on local time. Other logs are even written directly by the tasks.
But it is confusing from a user perspective that the time zones aren't shown in the same way.

Thinking about configs for this that could be used both by the server and by the tasks (e.g. the templates set up that tasks use this config) would be useful

**Comment by jluethi** [comment]
Alternatively, let's consider setting time zones?

**Comment by tcompa** [comment]
> Some of that could be handled through our new logging tooling in fractal-tasks-tools

This issue originally was about adapting the backend logs to a given timezone, rather than changing the tasks' log timestamps.

---

If the goal is to change both the tasks' logs timestamps _and_ the backend ones, then the timestamp information must be determined on the backend side. And in fact it should also be set at the systemd-services level, because we don't want e.g. backend/frontend logs in different time zones.

---

Ideally we should not play with `logging` configuration (as described in https://github.com/fractal-analytics-platform/fractal-server/issues/1694#issuecomment-3595742441), but rather with the `TZ` env variable - see e.g.
```console
$ TZ=Europe/Lisbon python3 -c 'import logging; logging.basicConfig(format="%(asctime)s %(levelname)8s %(message)s"); logging.warning("message")'
2026-03-23 07:32:50,240  WARNING message

$ TZ=Europe/Zurich python3 -c 'import logging; logging.basicConfig(format="%(asctime)s %(levelname)8s %(message)s"); logging.warning("message")'
2026-03-23 08:32:55,895  WARNING message
```

If the backend is aware of a given timezone string (e.g. `Europe/Zurich`), it can add `export TZ="Europe/Zurich"` to the SLURM submission script so that that all tasks' logs are emitted in this timezone.


---

To recap, here is one way to move forward (to be verified):
1. We select a specific timezone per instance (e.g. `Europe/Zurich` or an equivalent version of the same).
2. On the Fractal VM, we either change the timezone at the machine level or at least we make sure that some systemd services use it.
3. We include the `export TZ=...` string in all SLURM submission scripts.

What would remain "out of sync" would be e.g. the start/end times of SLURM jobs in `sacct`, but those are not exposed to users through the API. And perhaps the same applies to timestamps of `srun` errors from within the SLURM stderr files - TBD.
